File: python/2601_towers_of_hanoi.py
#!/usr/bin/python

import sys

class Towers:
    def print_move(self, fro, to):
        print 'Move from ' + str(fro) + ' to ' + str(to)

    def move(self, n, fro, to, spare):
        if n == 1:
            self.print_move(fro, to)
        else:
            self.move(n - 1, fro, spare, to)
            self.move(1, fro, to, spare)
            self.move(n - 1, spare, to, fro)

no_of_pegs = int(sys.argv[1])
towers = Towers()
towers.move(no_of_pegs, 'A', 'B', 'C')

File: ruby/2601_towers_of_hanoi.rb
#! /usr/bin/ruby

class Towers

    def move(n, fro, to, spare)
        if n <= 1
            print_move(fro, to)
        else
            move(n - 1, fro, spare, to)
            move(1, fro, to, spare)
            move(n - 1, spare, to, fro)
        end
    end

    private
    def print_move(fro, to)
        puts "Move peg from #{fro} to #{to}."
    end
end

towers_of_hanoi = Towers.new
ring_count = ARGV[0].to_i
towers_of_hanoi.move(ring_count, 'A', 'B', 'C')

